util/test/checkChplInstall

```diff
@@ -96,6 +96,30 @@ else
     CHPL_DIR_REMOVE=0
 fi
 
+# Tmp directory for compile output
+TMP_TEST_DIR="$(mktemp -d ${CHPL_DIR}/chapel-test-XXXXX)"
+log_info "Temporary test job directory: ${TMP_TEST_DIR}"
+
+# Cleanup the tmp directory whenever exit is invoked
+function cleanup_tmp_dir()
+{
+    local i
+    log_debug "Removing ${TMP_TEST_DIR}"
+    for i in 1 2 3 4 5
+    do
+        rm -rf ${TMP_TEST_DIR} && break
+        # sometimes rm needs to wait for the TEST_JOB exe file to close
+        log_debug "wait and try again"
+        sleep 1
+    done
+    if [ "${CHPL_DIR_REMOVE}" == "1" ]; then
+        log_debug "Removing ${CHPL_DIR}"
+        # rmdir should fail if some other process started using CHPL_DIR after we created it
+        rmdir ${CHPL_DIR} || log_debug "error ignored"
+    fi
+}
+trap cleanup_tmp_dir EXIT
```

Comment:
Moving the "trap" up here makes the clean_tmp_dir function fire if this script exits in the next ~45 lines.